src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -262,7 +299,16 @@ private void ProcessNewHeads()
                         }
                         catch (Exception e)
                         {
-                            if (_logger.IsDebug) _logger.Debug($"TxPool failed to update after block {args.Block.ToString(Block.Format.FullHashAndNumber)} with exception {e}");
+                            if (_logger.IsDebug)
+                                _logger.Debug(
+                                    $"TxPool failed to update after block {args.Block.ToString(Block.Format.FullHashAndNumber)} with exception {e}");
+                        }
+                        finally
+                        {
+                            if (--_headsQueueCount == 0)
```

Comment:
This is a non-atomic field that is updated from several places: `ProcessNewHeads` and `OnHeadChange`, why this should work?

Reply:
yeah true, I was thinking that single reader and writer is enough, but we do need to make it interlocked